tests/extmod/machine_disable_irq.py

```diff
@@ -0,0 +1,17 @@
+# Test executing Python code while IRQs disabled.
+try:
+    from machine import disable_irq, enable_irq
+    from time import ticks_us, ticks_add, ticks_diff
+except ImportError:
+    print("SKIP")
+    raise SystemExit
+
+for _ in range(5):
+    st_irq = disable_irq()
```

Comment:
Although it's not explicitly documented as such, the intention is that calls to this function can be nested.

I tested that with this PR on ESP32 and ESP32C3, and it seems to work, so that's great!

I suggest adding a test for that nested calling.  Eg:
```py
def f(n):
    st_irq = disable_irq()
    if n:
        f(n - 1)
    else:
        <tight loop code>
    enable_irq(st_irq)

for nest in range(3):
    for _ in range(5):
        f(nest)
```

Reply:
Done, have swapped test and confirmed it still fails before and passes after the fix is applied on C3.